ocm_python_wrapper/cluster.py

```diff
@@ -256,33 +257,85 @@ def addon_info(self):
             self.addon_name
         ).to_dict()
 
-    def validate_addon_parameters(self, parameters):
+    def filter_required_addon_parameters(self, parameters=None):
+        """Filters out addon parameter by specified cluster-related conditions."""
+
+        filtered_parameters = []
+        if parameters is None:
+            parameters = []
+
+        for param in parameters:
+            for param_condition in param["conditions"]:
+                if param_condition["resource"] == "cluster":
+                    validation_flag = True
+                    for condition in param_condition["data"]:
+                        condition_value = param_condition["data"][condition]
+                        cluster_value = benedict(
+                            self.instance.to_dict(), keypath_separator="."
+                        ).get(condition)
+                        # Using logical nor to check if cluster value doesn't meet all the condition requirements.
+                        if not (
+                            (
+                                isinstance(condition_value, list)
+                                and (cluster_value in condition_value)
+                            )
+                            or cluster_value == condition_value
+                        ):
+                            validation_flag = False
+                            break
+                    if validation_flag:
+                        filtered_parameters.append(param)
+        return filtered_parameters
+
+    def validate_addon_parameters(self, user_parameters, use_api_defaults):
```

Comment:
`use_api_defaults` - re-looking at the code, please set a default here (set to `True`)

Reply:
Done